src/Perscom/Service/AwardNominationService.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Forumify\PerscomPlugin\Perscom\Service;
+
+use Forumify\Core\Repository\SettingRepository;
+use Forumify\PerscomPlugin\Perscom\Entity\AwardNomination;
+use Forumify\PerscomPlugin\Perscom\Repository\AwardNominationRepository;
+use Forumify\PerscomPlugin\Perscom\Exception\AwardNominationAlreadyExistsException;
+
+class AwardNominationService
+{
+    public function __construct(
+        private readonly AwardNominationRepository $awardNominationRepository,
+        private readonly SettingRepository $settingRepository
+    ) {
+    }
+
+    public function create(AwardNomination $nomination): void
+    {
+        $this->ensureNotDuplicate($nomination);
+
+        $this->awardNominationRepository->save($nomination);
+    }
+
+    private function ensureNotDuplicate(AwardNomination $nomination): void
+    {
+        $existing = $this->awardNominationRepository->findBy([
+            'perscomUserId' => $nomination->getPerscomUserId(),
+            'receiverUserId' => $nomination->getReceiverUserId(),
+            'awardId' => $nomination->getAwardId(),
+            'status' => $this->settingRepository->get('perscom.award_nominations.pending_status_id')
+        ]);
+
+        if (!empty($existing)) {
+            throw new AwardNominationAlreadyExistsException();
+        }
```

Comment:
I think I am doing a similar thing somewhere, and seeing it now is kinda dummy 😂 

We can just use `->countBy` with the same arguments, and check `> 0`, a count is way more performant than doing an actual find and then not using the result haha